processor/PlanExecutor.go

```diff
@@ -86,12 +95,17 @@ func (instance *PlanExecutionContext) workerExecuteJob(ctx context.Context, job
 func (instance *PlanExecutionContext) executeStep(ctx context.Context, step core.Step) core.ExecutionResult {
 	start := time.Now()
 
+	//initilie the job contexts
 	if instance.JobContexts[step.JobID] == nil {
 		instance.JobContexts[step.JobID] = map[string]interface{}{}
 		instance.StepContexts[step.JobID] = map[int]core.ExtractionResult{}
 		instance.StepContexts[step.JobID][step.ID] = map[string]interface{}{}
 	}
 
+	if instance.StepContexts[step.JobID][step.ID] == nil {
+		instance.StepContexts[step.JobID][step.ID] = map[string]interface{}{}
```

Comment:
Really want to introduce types instead of all these unknown `map[string]interface{}{}`